File: src/main/java/com/indiepost/repository/PostReadingRepository.java
package com.indiepost.repository;

import com.indiepost.model.PostReading;

import java.util.List;

public interface PostReadingRepository {
    void save(PostReading postReading);

    Long save(PostReading postReading, Long userId, Long postId);

    PostReading findOne(Long id);

    PostReading findOneByUserIdAndPostId(Long userId, Long postId);

    List<PostReading> findByUserIdAndPostIds(Long userId, List<Long> postIds);

    void setVisibility(Long userId, boolean visible);

    void deleteById(Long id);
}

File: src/main/kotlin/com/indiepost/repository/PostReadingRepository.kt
package com.indiepost.repository

import com.indiepost.model.PostReading

interface PostReadingRepository {
    fun save(postReading: PostReading)

    fun save(postReading: PostReading, userId: Long, postId: Long): Long?

    fun findOne(id: Long?): PostReading?

    fun findOneByUserIdAndPostId(userId: Long, postId: Long): PostReading?

    fun findByUserIdAndPostIds(userId: Long, postIds: List<Long>): List<PostReading>

    fun setVisibility(userId: Long, visible: Boolean)

    fun deleteById(id: Long)
}
